### PR TITLE
[Driver][SYCL] Improve -std setting behaviors for Windows

### PR DESCRIPTION
**Pull request by mdtoguchi**
When using -std:arg on Windows, the std setting was not being applied
to the device compilation.  Make needed adjustements to allow this
which is due to the -std option differences in regards to Windows.

### CODE REVIEW

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 1]
```diff
@@ -5171,7 +5171,9 @@ void Clang::ConstructJob(Compilation &C, const JobAction &JA,
                                 /*Joined=*/true);
     else if (IsWindowsMSVC)
       ImplyVCPPCXXVer = true;
-    else if (IsSYCL)
+
+    if (IsSYCL && types::isCXX(InputType) &&
+        !Args.hasArg(options::OPT__SLASH_std))
```
Comment (AGindinson) [inline clang/lib/Driver/ToolChains/Clang.cpp:5176]: So is the difference in the behavior due to the MSVC driver not overriding the value when a new argument is encountered on the command line?

File: clang/test/Driver/sycl-offload.c [hunk 2]
```diff
@@ -891,7 +891,9 @@
 // -std=c++17 override check
 // RUN: %clangxx -### -c -fsycl -std=c++14 -xc++ %s 2>&1 | FileCheck -check-prefix=CHECK-STD-OVR %s
 // RUN: %clang_cl -### -c -fsycl /std:c++14 -TP %s 2>&1 | FileCheck -check-prefix=CHECK-STD-OVR %s
```
Comment (AGindinson) [inline clang/test/Driver/sycl-offload.c:893]: Does this test the branch with `IsMSVC2015Compatible` equaling `true`, or does the overriding scenario go down another code path? If the latter is correct, should we test the former as well?

File: clang/lib/Driver/ToolChains/Clang.cpp (hunk 1 above)
Comment (mdtoguchi) [reply clang/lib/Driver/ToolChains/Clang.cpp:5176]: Right .  We were setting up a default value (`-std=c++17`) for all platforms.  Due to the way the Windows MSVC path is taken (defaults set further down the line to better interact with `OPT__SLASH_std_EQ` we lost the ability to interject the user setting of `-std` and the default value was always set.

File: clang/test/Driver/sycl-offload.c (hunk 2 above)
Comment (mdtoguchi) [reply clang/test/Driver/sycl-offload.c:893]: Regardless of `IsMSVC2015Compatible` settings, `-std=c++17` will be used in the case where no `-std` is set by the user.  The only cases available for `-fsycl` is the user setting their value, or the default `-std=c++17`.  Thought about testing for `NOT -std=c++14` but that should be equivalent to checking for `-std=c++17`

**Review by AGindinson — APPROVED** [review]

**Pull request merged**